GPU/Common/SoftwareTransformCommon.cpp

```diff
@@ -856,16 +884,23 @@ void SoftwareTransform::ExpandLines(int vertexCount, int &maxIndex, u16 *&inds,
 		}
 	}
 
-	inds = newInds;
+	indsOffset = newIndsOffset;
+	return true;
 }
 
-void SoftwareTransform::ExpandPoints(int vertexCount, int &maxIndex, u16 *&inds, const TransformedVertex *transformed, TransformedVertex *transformedExpanded, int &numTrans, bool throughmode) {
+bool SoftwareTransform::ExpandPoints(int vertexCount, int &maxIndex, u16 *inds, int &indsOffset, int indexBufferSize, const TransformedVertex *transformed, TransformedVertex *transformedExpanded, int &numTrans, bool throughmode) {
+	// Before we start, do a sanity check - does the output fit?
+	if (vertexCount * 6 > indexBufferSize - indsOffset) {
+		// Won't fit, kill the draw.
+		return false;
+	}
+
 	numTrans = 0;
 	TransformedVertex *trans = &transformedExpanded[0];
 
-	const u16 *indsIn = (const u16 *)inds;
-	u16 *newInds = inds + vertexCount;
-	u16 *indsOut = newInds;
+	const u16 *indsIn = (const u16 *)(inds + indsOffset);
+	int newIndsOffset = indsOffset + vertexCount;
```

Comment:
Maybe it'd be cleaner for `result.indsBuffer` or something to exist, and be the resulting index pointer.  We already use `result.drawBuffer`, so it'd make the code more symmetrical and probably easier to understand.

-[Unknown]